schema_based_on_diagram.sql

```diff
@@ -0,0 +1,56 @@
+CREATE DATABASE clinic;
+
+CREATE TABLE patients(
+  id            INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  name          VARCHAR(50),
+  date_of_birth DATE
+);
+
+CREATE TABLE medical_histories(
+  id            INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  admitted_at TIMESTAMP,
+  status  VARCHAR(10),
+  patient_id INT,
+
+    CONSTRAINT fk_patient_id
+      FOREIGN KEY(patient_id) 
+	  REFERENCES patients(id)   
+);
+
+CREATE INDEX medical_histories_patient_id ON medical_histories(patient_id);
+
+CREATE TABLE treatments(
+  id            INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  type          VARCHAR(50),
+  name          VARCHAR(50)
+);
+
+
+CREATE TABLE invoices(
+  id            INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  total_amount          decimal,
+  generated_amount          TIMESTAMP,
+  payed_at          TIMESTAMP,
+  medical_history_id INT,
+
+  CONSTRAINT fk_medical_history_id
+      FOREIGN KEY(medical_history_id) 
+	  REFERENCES medical_histories(id)  
+
+);
+
+CREATE INDEX invoices_medical_history_id ON invoices(medical_history_id);
+
+CREATE TABLE invoice_items(
+  id            INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  unit_price    decimal,
+  total_price    decimal,
+  quantity      INT,
+  invoice_id     INT,
+
+  CONSTRAINT fk_invoice_id
+      FOREIGN KEY(invoice_id) 
+	  REFERENCES invoices(id)  
+);
+
+CREATE INDEX invoice_items_invoice_id ON invoice_items(invoice_id);
```

Comment:
- [ ] We are missing one more table: all "join table" for the many-to-many relastionships should be created, even if they are not showed on the diagram. There fore you should create a table between medical_histories and treatments  (name can be different)

Reply:
Thank you for your feedback @descholar-ceo. Changes will be implemented immediately.

Reply:
All the changes have been implemented.